runtime/src/cost_model.rs

```diff
@@ -200,6 +206,35 @@ impl CostModel {
         }
     }
 
+    /// eventually, potentially determine account data size of all writable accounts
+    /// at the moment, calculate account data size of account creation
+    fn calculate_account_data_size(&self, transaction: &SanitizedTransaction) -> u64 {
+        let mut sum = 0;
+        for (program_id, instruction) in transaction.message().program_instructions_iter() {
+            if program_id == &system_program::id() {
+                if let Ok(instruction) = limited_deserialize(&instruction.data) {
+                    let space = match instruction {
+                        SystemInstruction::CreateAccount {
+                            lamports: _lamports,
+                            space,
+                            owner: _owner,
+                        } => space,
+                        SystemInstruction::CreateAccountWithSeed {
+                            base: _base,
+                            seed: _seed,
+                            lamports: _lamports,
+                            space,
+                            owner: _owner,
+                        } => space,
+                        _ => 0,
```

Comment:
I think we also need to consider `SystemInstruction::Allocate{,WithSeed}` here, otherwise the checks can be sidestepped with a discrete `Allocate` -> `Assign` -> `Transfer`

Reply:
I added these 2 instructions.

Reply:
@jackcmay what about the realloc instruction?

Reply:
Realloc is not an instruction, it is an operation that a program can do on accounts they own, from within the program:
https://github.com/solana-labs/solana/blob/7da2df7d8c3960c0694513f65ff2d26c963d6acd/sdk/program/src/account_info.rs#L130

Reply:
> Realloc is not an instruction, it is an operation that a program can do on accounts they own, from within the program:
> 
> https://github.com/solana-labs/solana/blob/7da2df7d8c3960c0694513f65ff2d26c963d6acd/sdk/program/src/account_info.rs#L130

It doesn't seem like there's any runtime accounting of this operation?  It'd be nice to consider its contributions to storage changes

Reply:
Curious because I do not know this part of the code yet; Is tracking realloc possible to do here?

Reply:
We could probably add some kind of limit to the number of bytes realloc'd, not sure where that would come from